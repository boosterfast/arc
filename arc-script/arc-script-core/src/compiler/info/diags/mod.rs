pub mod repr;
pub mod to_codespan;
pub mod sink;
pub(crate) mod utils;

pub use repr::*;
pub use sink::*;
