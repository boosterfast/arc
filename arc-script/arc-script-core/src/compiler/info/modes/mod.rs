pub mod repr;
/// Module for representing sink and source information.
pub mod connector;

pub use repr::*;
