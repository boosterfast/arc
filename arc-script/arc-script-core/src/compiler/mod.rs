/// Module for representing Arc Queries.
pub(crate) mod query;
/// Module for representing the Abstract Syntax Tree.
///
/// NB: This module is public so that external libraries can construct ASTs
/// without having to generate source code.
pub mod ast;
/// Module for representing Dataflow graphs. Exports data types for staging
/// `HIR` functions.
pub mod dfg;
/// Module for representing Higher Order Intermediate Representation code.
pub(crate) mod hir;
/// Module for representing side information.
pub mod info;
/// Module for representing Multi-Level Intermediate Representation code.
pub(crate) mod mlir;
/// Module for representing Arcon code.
pub(crate) mod arcon;

// Module for incremental compilation.
// pub(crate) mod database;
/// Module which assembles compilation-pipeline.
pub mod pipeline;

pub(crate) mod pretty;

pub use pipeline::compile;
