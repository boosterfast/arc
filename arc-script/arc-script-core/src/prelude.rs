//! Exports for users who use the compiler as a library.

pub use crate::compiler;
pub use crate::compiler::ast;
pub use crate::compiler::info::diags;
pub use crate::compiler::info::logger;
pub use crate::compiler::info::modes;

pub use anyhow::Result;
pub use half::bf16;
pub use half::f16;
