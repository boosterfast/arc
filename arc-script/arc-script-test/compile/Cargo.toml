[package]
name        = "arc-script-test-compile"
version     = "0.0.0"
authors     = ["Klas Segeljakt <klasseg@kth.se>"]
edition     = "2018"
license     = "MIT"
description = "Tests to see if arc-scripts compile"

[dependencies]
trybuild        = { version = "1.0.39" }
arc-script-core = { version = "=0.0.0", path = "../../arc-script-core" }
insta           = { version = "1.6.0", features = ["glob"] }

[dev-dependencies]
arcon           = { git = "https://github.com/cda-group/arcon" }
arc-script      = { version = "=0.0.0", path = "../../" }
prost           = { version = "0.7.0" }
