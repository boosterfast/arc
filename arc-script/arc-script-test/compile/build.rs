fn main() {
    arc_script_build::Builder::default()
        .no_exclude(true)
        .source_dirs(["src/tests/expect_pass", "src/tests/expect_mlir_fail_todo"])
        .build();
}
