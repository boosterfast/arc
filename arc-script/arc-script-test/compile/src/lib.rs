#![allow(unused)]

#[cfg(test)]
mod insta;

// #[cfg(test)]
// mod trybuild;

// #[cfg(test)]
// mod compiletest;

mod tests {
    mod expect_pass {
        mod basic_pipe;
        mod binops;
        mod enum_pattern;
        mod enum_pattern_nested;
        mod enums;
        mod fib;
        mod fun;
        mod if_let;
        mod ifs;
        mod lambda;
        mod literals;
        mod nested_if;
        mod option;
        mod path;
        mod pattern;
        mod r#if;
        mod structs;
    }
    mod expect_mlir_fail_todo {
        mod extern_fun;
        mod pipe;
        mod stateful;
        mod task_filter;
        mod task_identity_untagged;
        mod task_map;
        mod task_with_funs;
    }
}
