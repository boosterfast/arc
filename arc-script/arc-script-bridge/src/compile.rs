use arc_script_core::compiler::compile;
use arc_script_core::compiler::info::diags::sink::Buffer;
use arc_script_core::prelude::modes::{Input, Mode, Output};

use proc_macro as pm;
use quote::quote;

use std::fs;
use std::path::PathBuf;

/// arc_script::compile!("script.arc")
///    .stage_fun1(1, 2, 3)
///    .stage_fun2("foo", "bar")
///    .finalize()
pub(crate) fn expand(input: pm::TokenStream) -> pm::TokenStream {
    todo!()
}
