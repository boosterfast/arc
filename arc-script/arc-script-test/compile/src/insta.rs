#![cfg_attr(not(test), allow(dead_code, unused_imports))]

use arc_script_core::prelude::compiler::compile;
use arc_script_core::prelude::diags::Buffer;
use arc_script_core::prelude::modes::{Input, Mode, Output};

use std::env;

fn snapshot(output: Output, paths: &str) {
    env::set_var("INSTA_OUTPUT", "summary");
    env::set_var("INSTA_FORCE_PASS", "1");
    let mut settings = insta::Settings::clone_current();
    settings.set_prepend_module_to_snapshot(false);
    settings.remove_snapshot_suffix();
    let mut sink = Buffer::no_color();
    settings.bind(|| {
        insta::glob!(paths, |path| {
            println!("Testing {}", path.display());
            let mode = Mode {
                input: Input::File(Some(path.into())),
                output,
                ..Default::default()
            };
            compile(mode, &mut sink).unwrap();
            let s = std::str::from_utf8(sink.as_slice()).unwrap();
            insta::assert_snapshot!(s);
            sink.clear();
        })
    });
}

#[test]
fn test_ast() {
    snapshot(Output::AST, "tests/expect_mlir_fail_todo/*.arc");
    snapshot(Output::AST, "tests/expect_pass/*.arc");
    snapshot(Output::AST, "tests/expect_fail/*.arc");
    snapshot(Output::AST, "tests/expect_fail_todo/*.arc");
}

#[test]
fn test_hir() {
    snapshot(Output::HIR, "tests/expect_mlir_fail_todo/*.arc");
    snapshot(Output::HIR, "tests/expect_pass/*.arc");
    snapshot(Output::HIR, "tests/expect_fail/*.arc");
    snapshot(Output::HIR, "tests/expect_fail_todo/*.arc");
}

#[test]
fn test_dfg() {
    snapshot(Output::DFG, "tests/expect_mlir_fail_todo/*.arc");
    snapshot(Output::DFG, "tests/expect_pass/*.arc");
    snapshot(Output::DFG, "tests/expect_fail/*.arc");
    snapshot(Output::DFG, "tests/expect_fail_todo/*.arc");
}

#[test]
fn test_rust() {
    snapshot(Output::Rust, "tests/expect_mlir_fail_todo/*.arc");
    snapshot(Output::Rust, "tests/expect_pass/*.arc");
    snapshot(Output::Rust, "tests/expect_fail/*.arc");
    snapshot(Output::Rust, "tests/expect_fail_todo/*.arc");
}

#[test]
fn test_mlir() {
    snapshot(Output::MLIR, "tests/expect_pass/*.arc");
    snapshot(Output::MLIR, "tests/expect_fail/*.arc");
    snapshot(Output::MLIR, "tests/expect_fail_todo/*.arc");
}
