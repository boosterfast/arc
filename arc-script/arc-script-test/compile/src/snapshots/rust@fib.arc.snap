---
source: arc-script-test/compile/src/insta.rs
expression: s
input_file: arc-script-test/compile/src/tests/expect_pass/fib.arc

---
#[allow(non_snake_case)]
#[allow(unused_must_use)]
#[allow(dead_code)]
#[allow(unused_variables)]
#[allow(unused_imports)]
#[allow(unused_braces)]
#[allow(irrefutable_let_patterns)]
#[allow(clippy::redundant_field_names)]
#[allow(clippy::unused_unit)]
#[allow(clippy::double_parens)]
pub mod arc_script_output {
    use super::*;
    use arc_script::arcorn;
    use arc_script::arcorn::*;
    pub fn fib(n_0: i32) -> i32 {
        let x_0: i32 = 2i32;
        let x_1: bool = (n_0.clone()) > (x_0.clone());
        let x_C: i32 = if (x_1.clone()) {
            let x_2: Box<dyn arcorn::ArcornFn(i32) -> i32> =
                Box::new(fib) as Box<dyn arcorn::ArcornFn(i32) -> i32>;
            let x_3: i32 = 1i32;
            let x_4: i32 = (n_0.clone()) - (x_3.clone());
            let x_5: i32 = (x_2.clone())((x_4.clone()));
            let x_6: Box<dyn arcorn::ArcornFn(i32) -> i32> =
                Box::new(fib) as Box<dyn arcorn::ArcornFn(i32) -> i32>;
            let x_7: i32 = 2i32;
            let x_8: i32 = (n_0.clone()) - (x_7.clone());
            let x_9: i32 = (x_6.clone())((x_8.clone()));
            let x_A: i32 = (x_5.clone()) + (x_9.clone());
            (x_A.clone())
        } else {
            let x_B: i32 = 0i32;
            (x_B.clone())
        };
        (x_C.clone())
    }
    pub fn test() -> i32 {
        let x_D: Box<dyn arcorn::ArcornFn(i32) -> i32> =
            Box::new(fib) as Box<dyn arcorn::ArcornFn(i32) -> i32>;
        let x_E: i32 = 5i32;
        let x_F: i32 = (x_D.clone())((x_E.clone()));
        (x_F.clone())
    }
}
pub use arc_script_output::*;


